services/wallet/bridge/bridge.go

```diff
@@ -100,19 +119,55 @@ func (t *TransactionBridge) Data() types.HexBytes {
 	return types.HexBytes("")
 }
 
+type BridgeParams struct {
+	FromChain *params.Network
+	ToChain   *params.Network
+	FromAddr  common.Address
+	ToAddr    common.Address
+	FromToken *token.Token
+	ToToken   *token.Token
+	AmountIn  *big.Int
+
+	// extra params
+	BonderFee *big.Int
+	Username  string
+	PublicKey string
+}
+
 type Bridge interface {
 	// returns the name of the bridge
 	Name() string
 	// checks if the bridge is available for the given networks/tokens
-	AvailableFor(from *params.Network, to *params.Network, token *token.Token, toToken *token.Token) (bool, error)
+	AvailableFor(params BridgeParams) (bool, error)
 	// calculates the fees for the bridge and returns the amount BonderFee and TokenFee (used for bridges)
-	CalculateFees(from, to *params.Network, token *token.Token, amountIn *big.Int) (*big.Int, *big.Int, error)
+	CalculateFees(params BridgeParams) (*big.Int, *big.Int, error)
 	// Pack the method for sending tx and method call's data
-	PackTxInputData(contractType string, fromNetwork *params.Network, toNetwork *params.Network, from common.Address, to common.Address, token *token.Token, amountIn *big.Int) ([]byte, error)
-	EstimateGas(fromNetwork *params.Network, toNetwork *params.Network, from common.Address, to common.Address, token *token.Token, toToken *token.Token, amountIn *big.Int) (uint64, error)
-	CalculateAmountOut(from, to *params.Network, amountIn *big.Int, symbol string) (*big.Int, error)
+	PackTxInputData(params BridgeParams, contractType string) ([]byte, error)
+	EstimateGas(params BridgeParams) (uint64, error)
+	CalculateAmountOut(params BridgeParams) (*big.Int, error)
 	Send(sendArgs *TransactionBridge, verifiedAccount *account.SelectedExtKey) (types.Hash, error)
-	GetContractAddress(network *params.Network, token *token.Token) (common.Address, error)
+	GetContractAddress(params BridgeParams) (common.Address, error)
 	BuildTransaction(sendArgs *TransactionBridge) (*ethTypes.Transaction, error)
-	BuildTx(fromNetwork, toNetwork *params.Network, fromAddress common.Address, toAddress common.Address, token *token.Token, amountIn *big.Int, bonderFee *big.Int) (*ethTypes.Transaction, error)
+	BuildTx(params BridgeParams) (*ethTypes.Transaction, error)
+}
+
+func extractCoordinates(pubkey string) ([32]byte, [32]byte) {
```

Comment:
nit: why are those 2 function inside the bridge.go, i would put them into a dedicated file?

Reply:
Yes, copied from there, will find a better place later for all of them. Most likely put them in some helper file or so. I have that in mind.